src/query/api/v1/httpd/handler.go

```diff
@@ -102,13 +105,19 @@ func NewHandler(
 ) (*Handler, error) {
 	r := mux.NewRouter()
 
-	// apply middleware. Just CORS for now, but we could add more here as needed.
-	withMiddleware := &cors.Handler{
+	// apply middleware.
```

Comment:
nit: maybe change comment to `// Add CORS to endpoint middleware`?

Reply:
Ah yeah I meant the comment to be "apply all middleware". I'll just factor out a function.